ComSemApp/models.py

```diff
@@ -343,6 +344,31 @@ def __str__(self):
 
     class Meta:
         verbose_name = "Speaking Practice Attempt"
+    
+    def review_requested(self) -> bool:
+        """
+            Returns true if there is a teacher review request for
+            this attempt
+        """
+        return hasattr(self,'request')
+
+    def teacher_reviewed(self) -> bool:
+        """
+            Returns true if there is a request and a review for the request
+            for this attempt
+        """
+        if self.review_requested():
+            return SpeakingPracticeAttemptReviewRequest.objects.get(attempt=self).is_reviewed()
+        return False
+
+    def get_review(self) -> Optional[SpeakingPracticeAttemptReview]:
```

Comment:
I'm not sure how much I like the `Optional` type hint. I personally prefer `SpeakingPracticeAttemptReview | None`